src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/Record2SolutionMappingTranslatorImpl.java

```diff
@@ -0,0 +1,129 @@
+                if (builders.isEmpty())
+                    baseBuilder.add(mappingVar, edgeMapping);
+                else {
+                    addToAllBuilders(builders, mappingVar, edgeMapping);
+                }
+            } else if (value instanceof MapValue) {
+                final Map<String, Object> mapValue = ((MapValue) value).getMap();
+                final Node triple = NodeFactory.createTripleNode(
+                        conf.mapNode((LPGNode) mapValue.get("source")),
+                        conf.mapEdgeLabel(((LPGEdge) mapValue.get("edge")).getLabel()),
+                        conf.mapNode((LPGNode) mapValue.get("target"))
+                );
+                if (builders.isEmpty())
+                    baseBuilder.add(mappingVar, triple);
+                else {
+                    addToAllBuilders(builders, mappingVar, triple);
+                }
+            } else if (value instanceof ListValue) {
+                final List<Object> values = ((ListValue) value).getList();
+                final boolean isPropertyList = CypherUtils.isPropertyColumn(query, var);
+                if (builders.isEmpty()) {
+                    for (final Object val : values) {
+                        Node node;
+                        if (isPropertyList) {
+                            node = conf.mapProperty(val.toString());
+                        } else {
+                            node = NodeFactory.createLiteral(val.toString());
+                        }
+                        final BindingBuilder builder = Binding.builder();
+                        builder.addAll(baseBuilder.build());
+                        builder.add(mappingVar, node);
+                        builders.add(builder);
+                    }
+                } else {
+                    for (final Object val : values) {
+                        Node node;
+                        if (isPropertyList) {
+                            node = conf.mapProperty(val.toString());
+                        } else {
+                            node = NodeFactory.createLiteral(val.toString());
+                        }
+                        addToAllBuilders(builders, mappingVar, node);
+                    }
+                }
+            } else if (value instanceof LiteralValue) {
+                final String literal = ((LiteralValue) value).getValue().toString();
+                Node node;
+                if (literal.equals("label"))
+                    node = conf.getLabel();
+                else
+                    node = NodeFactory.createLiteral(literal);
+                if (builders.isEmpty()) {
+                    baseBuilder.add(mappingVar, node);
+                } else {
+                    addToAllBuilders(builders, mappingVar, node);
+                }
+            } else {
+                throw new IllegalArgumentException("Unsupported implementation of Value");
```

Comment:
```suggestion
                throw new IllegalArgumentException("Unsupported implementation of value (" + value.getClass().getName() + ")");
```
Including the name of the unexpected type helps if this issue actually occurs ;-)